evalml/data_checks/target_leakage_data_check.py

```diff
@@ -1,69 +1,70 @@
 """Data check that checks if any of the features are highly correlated with the target by using mutual information or Pearson correlation."""
+from woodwork.config import CONFIG_DEFAULTS
+
 from evalml.data_checks import (
     DataCheck,
     DataCheckActionCode,
     DataCheckActionOption,
     DataCheckMessageCode,
     DataCheckWarning,
 )
-from evalml.utils.woodwork_utils import infer_feature_types, numeric_and_boolean_ww
+from evalml.utils.woodwork_utils import infer_feature_types
 
 
 class TargetLeakageDataCheck(DataCheck):
-    """Check if any of the features are highly correlated with the target by using mutual information or Pearson correlation.
+    """Check if any of the features are highly correlated with the target by using mutual information, Pearson correlation, and other correlation metrics.
 
-    If `method='mutual'`, this data check uses mutual information and supports all target and feature types.
-    Otherwise, if `method='pearson'`, it uses Pearson correlation and only supports binary with numeric and boolean dtypes.
-    Pearson correlation returns a value in [-1, 1], while mutual information returns a value in [0, 1].
+    If `method='mutual_info'`, this data check uses mutual information and supports all target and feature types.
+    Other correlation metrics only support binary with numeric and boolean dtypes, and return a value in [-1, 1], while mutual information returns a value in [0, 1].
+    Correlation metrics available can be found in Woodwork's `dependence_dict method <https://woodwork.alteryx.com/en/stable/generated/woodwork.table_accessor.WoodworkTableAccessor.dependence_dict.html#woodwork.table_accessor.WoodworkTableAccessor.dependence_dict>`_.
 
     Args:
         pct_corr_threshold (float): The correlation threshold to be considered leakage. Defaults to 0.95.
-        method (string): The method to determine correlation. Use 'mutual' for mutual information, otherwise 'pearson' for Pearson correlation. Defaults to 'mutual'.
+        method (string): The method to determine correlation. Use 'max' for the maximum correlation, or for specific correlation metrics, use their name (ie 'mutual_info' for mutual information, 'pearson' for Pearson correlation, etc).
+            Defaults to 'mutual_info'.
```

Comment:
Default to `mutual_info` for now as we [fix a bug](https://github.com/alteryx/woodwork/pull/1534) in WW

Reply:
Should we consider this a breaking change, since `mutual` is no longer accepted as an input to this Data Check?